third_party/lint/0001-enable-buildifier-test-without-sandbox.patch

```diff
@@ -0,0 +1,137 @@
+diff --git a/buildifier/buildifier.bzl b/buildifier/buildifier.bzl
+index 42445e7..69f3341 100644
+--- a/buildifier/buildifier.bzl
++++ b/buildifier/buildifier.bzl
+@@ -34,8 +34,24 @@ def buildifier(**kwargs):
+ def _buildifier_test_impl(ctx):
+     return [buildifier_impl_factory(ctx, test_rule = True)]
+ 
+-buildifier_test = rule(
++_buildifier_test = rule(
+     implementation = _buildifier_test_impl,
+     attrs = buildifier_attr_factory(True),
+     test = True,
+ )
++
++def buildifier_test(**kwargs):
++    """
++    Wrapper for the _buildifier_test rule. Adds tags to disable sandboxing.
++
++    Args:
++      **kwargs: all parameters for _buildifier_test
++    """
++    if kwargs.get("no-sandbox", False):
++        tags = kwargs.get("tags", [])
++        # Note: the "external" tag is a workaround for bazelbuild#15516.
++        for t in ["no-sandbox", "no-cache", "external"]:
++            if t not in tags:
++                tags.append(t)
++        kwargs["tags"] = tags
++    _buildifier_test(**kwargs)
+diff --git a/buildifier/internal/factory.bzl b/buildifier/internal/factory.bzl
+index 779c789..0ae3bc1 100644
+--- a/buildifier/internal/factory.bzl
++++ b/buildifier/internal/factory.bzl
+@@ -46,6 +46,10 @@ def buildifier_attr_factory(test_rule = False):
+         "verbose": attr.bool(
+             doc = "Print verbose information on standard error",
+         ),
++        "exclude_patterns": attr.string_list(
++            allow_empty = True,
++            doc = "A list of glob patterns passed to the find command. E.g. './vendor/*' to exclude the Go vendor directory. In test rules, this attribute requires the use of the no_sandbox attribute.",
++        ),
+         "mode": attr.string(
+             default = "fix" if not test_rule else "diff",
+             doc = "Formatting mode",
+@@ -80,7 +84,7 @@ def buildifier_attr_factory(test_rule = False):
+     if test_rule:
+         attrs.update({
+             "srcs": attr.label_list(
+-                allow_empty = False,
++                allow_empty = True,
+                 allow_files = [
+                     ".bazel",
+                     ".bzl",
+@@ -91,12 +95,13 @@ def buildifier_attr_factory(test_rule = False):
+                 ],
+                 doc = "A list of labels representing the starlark files to include in the test",
+             ),
+-        })
+-    else:
+-        attrs.update({
+-            "exclude_patterns": attr.string_list(
+-                allow_empty = True,
+-                doc = "A list of glob patterns passed to the find command. E.g. './vendor/*' to exclude the Go vendor directory",
++            "no_sandbox": attr.bool(
```

Comment:
Do we need this attr?  Would it be easier to make existence of a non-empty `workspace` label imply the behavior currently encoded by `no-sandbox`?

Reply:
I'm open to either way. My thinking on this was that enabling this no-sandbox behavior may have implications for supporting remote execution in the future, so I wanted to be very clear in the attribute about the fact that we are exiting the sandbox for this rule (otherwise you'd have to read the source code for buildifer to figure this out). It does make the code a bit more complicated as we have to check another condition.

I'm happy to remove it though if we don't think it's worth it.

Reply:
Can we keep it for now and use whatever we can upstream in the long run? Or is your thinking that we'll have to do this sandbox escape for all the tools and we should keep it concise?

Reply:
The existing sandbox-escaping tests have been written to use the `workspace` attr as the sandbox-escape indicator (it tells the script which directory to go to).  If this turns out to be a problem for remote exec, we'll have to fix it in several places.

Reply:
Got it - how about we keep it as is for now, and I attempt to upstream and see what the maintainers say.

Reply:
SGTM.